src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -304,21 +269,21 @@ public async Task StoreAsync(object entity, string id, IMetadataDictionary metad
                     _first = false;
                     _inProgressCommand = CommandType.None;
 
-                    _currentWriter.Write("{\"Id\":\"");
+                    _writer.Write("{\"Id\":\"");
                     WriteString(id);
-                    _currentWriter.Write("\",\"Type\":\"PUT\",\"Document\":");
+                    _writer.Write("\",\"Type\":\"PUT\",\"Document\":");
 
-                    _currentWriter.Flush();
+                    await _writer.StreamWriter.FlushAsync().ConfigureAwait(false);
```

Comment:
why not to have `_writer.FlushAsync()` ?

Reply:
Yes. Done